src/ASPHERE/pair_gayberne.cpp

```diff
@@ -531,10 +531,14 @@ void PairGayBerne::read_restart_settings(FILE *fp)
 void PairGayBerne::write_data(FILE *fp)
 {
   for (int i = 1; i <= atom->ntypes; i++)
-    fprintf(fp,"%d %g %g %g %g %g %g %g %g\n",i,
+    if (!atom->chartypesflag) fprintf(fp,"%d %g %g %g %g %g %g %g %g\n",i,
             epsilon[i][i],sigma[i][i],
             pow(well[i][0],-mu),pow(well[i][1],-mu),pow(well[i][2],-mu),
             pow(well[i][0],-mu),pow(well[i][1],-mu),pow(well[i][2],-mu));
+    else fprintf(fp,"%d %g %g %g %g %g %g %g %g # %s\n",i,
+            epsilon[i][i],sigma[i][i],
+            pow(well[i][0],-mu),pow(well[i][1],-mu),pow(well[i][2],-mu),
+            pow(well[i][0],-mu),pow(well[i][1],-mu),pow(well[i][2],-mu),atom->char_atomtype[i-1]);
```

Comment:
does this mean, that `Atom::char_atomtype[]` is 0-based and not 1-based like practically all other atom type based arrays?

Reply:
will correct this